src/Compilers/Core/Portable/Text/SourceText.cs

```diff
@@ -942,28 +943,28 @@ protected virtual TextLineCollection GetLinesCore()
         internal sealed class LineInfo : TextLineCollection
         {
             private readonly SourceText _text;
-            private readonly int[] _lineStarts;
+            private readonly SegmentedList<int> _lineStarts;
```

Comment:
💡 This could likely be changed to an `ImmutableSegmentedList<int>` with minimal penalty. Note that there is not _currently_ a way to create the builder for that with a specific capacity, but I'm already working on a change to add that.